examples/music-licensing-challenge/src/app/api/movies.py

```
@@ -0,0 +1,24 @@
+from typing import List, Optional
+from ..db.database import get_db
+from ..models.movie import Movie
+from ..schemas.movies import MovieWithAllData
+from ..repository.movies import MovieRepository
+
+router = APIRouter()
+
+
+@router.get("/", response_model=List[MovieWithAllData] | MovieWithAllData)
+def read_movies(
+    id: Optional[str] = None,
```

Ruff (0.15.13)

[error] Function argument `id` is shadowing a Python builtin

(A002)